src/plugins/supply-demand/languages/BasicJS/starter-code/main.js

```diff
@@ -1,18 +1,30 @@
-// Base curves are created automatically when you use shift functions!
+//law of supply demo
+for (let i = -0.8; i < 0.8; i+= 0.0001) {
+     setPrice(i)
+     setQuantity(i)
+     }
+     
+     
+     
 
-// Shift the curves using simple functions
-addSupply(0.5);  // Shift supply curve right by 0.5
-addDemand(0.3);    // Shift demand curve right by 0.3
-addDrought(5);    // Shift supply curve left by 50 (5 * 10)
-
-// Show equilibrium point and helper lines
-showHelpers(); 
-
-
-// Set price and quantity
-setPrice(0.5);
-setQuantity(0.5);
+     //law of demand demo:
+     for (let i = -0.8; i < 0.8; i+= 0.0001) {
+     setPrice(-i)
+     setQuantity(i)
+     }
+     
+     
+     for (let i =0; i < 0.8; i +=0.0001) {
+     addDemand(0.0001)
+     }
+     
 
+     for (let i = 0; i < 1; i+= 0.1) {
+          setPrice(i)
+          setQuantity(i)
+          draw()
```

Comment:
Why do we only draw here in this example? Also I'm not sure about the loop bounds again